packages/compiler/core/checker.ts

```diff
@@ -1754,12 +1753,21 @@ export function createChecker(program: Program): Checker {
     return getMergedSymbol(sym);
   }
 
-  function reportAmbiguousIdentifier(node: IdentifierNode, symbols: Sym[]) {
-    const duplicateNames = symbols.map(getFullyQualifiedSymbolName).join(", ");
+  function reportAmbiguousIdentifier(
+    node: IdentifierNode,
+    symbols: Sym[],
+    prefixForEmpty?: string
+  ) {
+    let duplicateNames = symbols.map(getFullyQualifiedSymbolName);
+    if (prefixForEmpty) {
+      duplicateNames = duplicateNames.map((name) =>
+        name.includes(".") ? name : `${prefixForEmpty}.${name}`
```

Comment:
Well, now we are checking for "." to decide whether to add the prefix, and the only case where we push a prefix is when we have already verified that it should be global.

I could move the logic to check if something is global into here, but it would just be duplicating the logic elsewhere that's already been done to verify that (and needs to continue to be there) so it seemed unnecessary to replicate it here.

Alternatively, `getFullyQualifiedSymbolName` should return with the `global.` prefix, which would accomplish the same thing, but I figured that could have more ramifications than here were we are simply formatting an error string.

Reply:
I guess I am not fully understanding that need, wouldn't the prefix always be `global.`

My idea here was have this logic:
1. you start with the list of symbols passed to `reportAmbiguousSymbol`
2. You call map so each symbol can be mapped  and get those 2 info  `[getFullyQualifiedSymbolName(symbol), isGlobal(symbol)]` where `isGlobal` check the `symbol.parent` which would replace a check for the `includes(".")` (which is in my opinion is less robust and slower)
3. After that you can rejoin as you have

Reply:
`symbol.parent` is not sufficient logic to distinguish that something is global. I tried this and it just appends global to everything, including fully-qualified names.

There's no other prefix I can think of besides global that makes, but I did that so that the current invocation of `reportAmiguousIdentifier` wouldn't inadvertently append global and only the code path which has already verified that the symbol is global will do so.

Even if I changed it from a prefix string to a `appendGlobal` boolean, the logic to determine that global should be appended will still rely on looking for the absence of ".". In regards to it being slower, IMO that doesn't matter since this is only used to create a diagnostic error string.

To implement an `isGlobal` I would need to modify the logic that already exists at line 2002:
```
      const globalBinding = resolveIdentifierInTable(
        node,
        globalNamespaceNode.symbol.exports,
        resolveDecorator
      );
```
Essentially I would need to do a reverse lookup on the globalNamespaceNode exports to see that the symbol is in the SymbolTable. That seems needlessly complicated just to create an error string and it seems to me that having to iterate through values instead of a simple key lookup would negate any potential speed savings.

Reply:
This should work. Then pass `{ useGlobalPrefixAtTopLevel: true }` in reportAmbiguousIdentier. That's not the best name for the option, though, since it means "only put global on unparented symbol, not root of all symbols." Open to better names and probably worth a doc comment.

```ts
  function getFullyQualifiedSymbolName(
    sym: Sym | undefined,
    options?: {
      useGlobalPrefixAtTopLevel?: boolean;
    }
  ): string {
    if (!sym) return "";
    if (sym.symbolSource) sym = sym.symbolSource;
    const parent =
      sym.parent && !(sym.parent.flags & SymbolFlags.SourceFile) ? sym.parent : undefined;
    const name = sym.flags & SymbolFlags.Decorator ? sym.name.slice(1) : sym.name;

    if (parent?.name) {
      return `${getFullyQualifiedSymbolName(parent)}.${name}`;
    } else if (options?.useGlobalPrefixAtTopLevel) {
      return `global.${name}`;
    } else {
      return name;
    }
  }
```

Reply:
I do think it will be easier on maintenance if caller doesn't need to pass isGlobal.

Reply:
I like this approach--I'll switch to that.

Reply:
I made this switch @timotheeguerin @nguerrera